### DISCUSSION TITLE
findFirst query takes in: skip, take

### DISCUSSION THREAD

**Discussion by nikola418** [prisma/prisma#20697] [Q&A] [answered]
### Question

Why does findFirst take in skip and take options when intrinsically it should take one and skip none

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @nikola418 👋 

The `findFirst` method in Prisma Client indeed takes in `skip` and `take` options, even though it is designed to return the first record that matches the specified conditions.

The `skip` option is used to specify how many of the returned objects in the list should be skipped. This can be useful when you want to ignore a certain number of records before selecting the first one that matches your conditions.

The `take` option, on the other hand, specifies how many objects should be returned in the list. When used with `findFirst`, take is implicitly 1 or -1. `findFirst` is only affected by whether the value is positive or negative - any negative value reverses the list.
So, while it might seem counterintuitive, these options provide additional flexibility when querying your data with findFirst. See the [documentation](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findfirst).


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
